package/src/core/define-integration.ts

```diff
@@ -43,7 +45,16 @@ export const defineIntegration = <
 	}) => ExtendedHooks<TPlugins>;
 }): ((options?: z.input<TOptionsSchema>) => AstroIntegration) => {
 	return (_options: z.input<TOptionsSchema> = {}) => {
-		const options = optionsSchema?.parse(_options) as z.output<TOptionsSchema>;
+		const parsedOptions = optionsSchema?.safeParse(_options, { errorMap });
+
+		if (parsedOptions && !parsedOptions.success) {
+			throw new AstroError(
+				`Invalid configuration passed to '${name}' integration\n`,
+					parsedOptions.error.issues.map((i) => i.message).join('\n')
+			);
+		}
+
+		const options = (parsedOptions?.data || _options) as z.output<TOptionsSchema>
```

Comment:
Currently in v0.2.0 if an `optionsSchema` is not defined the integration will still run and not throw an error, the `|| _options` is meant to handle that case but I am all for removing it and requiring an `optionsSchema` or maybe a default of something like `z.record(z.unknown())`

Reply:
I have not strong opinions about this, @jdtjenkins @Fryuni what are your thoughts?

Reply:
If no schema is passed, it will accept anything since it should default to `AnyZodObject`, both for the integration input and the setup input (output of the validation).

Being explicit with a default to `z.record(z.unknown())` is preferable, but also a breaking change. The current typing will default to `Record<string, any>` when a schema is not passed in, which exits the type checking and allows the use of any value. `z.record(z.unknown())` becomes `Record<string, unknown>` which requires an explicit cast, check or type guard before using any property in the options.

Safer, but also way more annoying.

Reply:
i'm not concerned about breaking changes at this stage, i guess safer is better!

Reply:
I think I agree with Fryuni, I tried typing the `options` object as `Record<string, unknown>` and it can be annoying to use and some of the typing would have to be redone. For example:

```ts
const { id } = options as unknown as Options
```

If the `options` objects needs a more narrow type of `Record<string, unknown>` it might be better suited for another PR



I just pushed some changes to clean up these conditionals by adding a default value to the `optionsSchema` argument